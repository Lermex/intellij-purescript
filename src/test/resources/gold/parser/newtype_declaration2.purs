module A where

newtype User3 = NoOp2
