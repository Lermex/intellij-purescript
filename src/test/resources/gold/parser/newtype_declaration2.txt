Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSProperNameImpl(ModuleName)
        PSProperNameImpl(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSNewTypeDeclarationImpl(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSIdentifierImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User3')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSIdentifierImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('NoOp2')