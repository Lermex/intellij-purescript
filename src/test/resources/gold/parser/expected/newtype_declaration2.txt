Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PsiElement(newtype)('newtype')
      PsiElement(whitespace)(' ')
      PsiElement(proper name)('User3')
      PsiElement(whitespace)(' ')
      PsiElement(=)('=')
      PsiElement(whitespace)(' ')
      PsiElement(proper name)('NoOp2')
      PsiErrorElement:Expecting proper name or TypeAtom
        <empty list>